src/AzureClient/EntryPoint/EntryPoint.cs

```diff
@@ -73,12 +84,78 @@ public Task<IQuantumMachineJob> SubmitAsync(IQuantumMachine machine, AzureSubmis
                 }
             }
 
-            var entryPointInput = parameterValues.Count switch
+            return parameterValues.Count switch
             {
                 0 => QVoid.Instance,
                 1 => parameterValues.Single(),
                 _ => InputType.GetConstructor(parameterTypes.ToArray()).Invoke(parameterValues.ToArray())
             };
+        }
+
+        private ArgumentValue GetArgumentValue(string parameterValue, System.Reflection.ParameterInfo parameter)
+        {
+            var parameterType = parameter.ParameterType;
+
+            if (parameterType == typeof(bool))
+            {
+                return new ArgumentValue.Bool(Newtonsoft.Json.JsonConvert.DeserializeObject<bool>(parameterValue));
+            }
+            else if (parameterType == typeof(double))
+            {
+                return new ArgumentValue.Double(Newtonsoft.Json.JsonConvert.DeserializeObject<double>(parameterValue));
+            }
+            else if (parameterType == typeof(long))
+            {
+                return new ArgumentValue.Int(Newtonsoft.Json.JsonConvert.DeserializeObject<long>(parameterValue));
+            }
+            else if (parameterType == typeof(string))
+            {
+                return new ArgumentValue.String(parameterValue);
+            }
+            else if (parameterType == typeof(Pauli))
+            {
+                return new ArgumentValue.Pauli(Newtonsoft.Json.JsonConvert.DeserializeObject<Pauli>(parameterValue));
+            }
+            else if (parameterType == typeof(Result))
+            {
+                return new ArgumentValue.Result(Newtonsoft.Json.JsonConvert.DeserializeObject<Result>(parameterValue)!);
+            }
+            else
+            {
+                throw new ArgumentException($"The given type of {parameterType.Name} is not supported."); ;
+            }
+        }
+
+        private IReadOnlyList<Argument> GetEntryPointInputArguments(AzureSubmissionContext submissionContext)
+        {
+            var argumentList = new List<Argument>();
+            foreach (var parameter in OperationInfo.RoslynParameters)
+            {
+                if (!submissionContext.InputParameters.ContainsKey(parameter.Name))
+                {
+                    throw new ArgumentException($"Required parameter {parameter.Name} was not specified.");
+                }
+
+                var rawParameterValue = submissionContext.InputParameters[parameter.Name];
+
+                try
+                {
+                    var argument = new Argument(parameter.Name, GetArgumentValue(rawParameterValue, parameter));
+                    argumentList.Add(argument);
+                }
+                catch (Exception e)
+                {
+                    throw new ArgumentException($"The value {rawParameterValue} provided for parameter {parameter.Name} could not be converted to the expected type: {e.Message}");
+                }
+            }
```

Comment:
(I can be wrong)
I have an impression that if we do the job submission to the AQ service, e.g. in command line using the Azure CLI, or in the Jupyter notebook, etc., and we forget to specify the required parameter, or we specify a wrong-type value (e.g. for an integer param we pass a string arg), then the `ArgumentException` will be thrown by this function, that exception will not be caught by any code, and the running code will _crash_ (after [partially] printing or not printing the exception message). Whereas it makes sense to gracefully fail with a message in such cases.

Reply:
In the worst case, C# exceptions during magic command execution are caught by `ExecuteMagic` and reported to the user without crashing the kernel; it is often better and more direct to use `IChannel` and `ExecutionResult` together, but exceptions are not inherently unsafe in this context.